### DISCUSSION TITLE
Communication between multiple instances

### DISCUSSION THREAD

**Discussion by kalebscholes** [wailsapp/wails#2441] [General]
I'm trying to get setup an application to use OAuth2 with a custom protocol handler. My issue at the moment is that once I get redirected back, the app creates a new instance and I now have two windows open. (I understand that v2 doesn't support multiple windows but wasn't sure if this fell into the same bucket?)  Is there a way to communicate between the two windows or is the suggested method to spin up a server and listen on a local loopback port?

**Comment by leaanthony** [comment]
That is the current recommendation and used by a number of projects 👍

**Comment by mstura** [comment]
While the http loopback works it does prompt the user on windows to allow the application to open a connection which might not always be desirable. Wouldn't using a combination of a custom protocol combined with the SingleInstanceLock work for this situation?
